trajectory_executor/com_serial.py

```diff
@@ -0,0 +1,222 @@
+          data_byte_counter -= 1
+          if data_byte_counter == 0:
+            data_byte_counter = self.data_byte_length
+            read_state = serial_read_states['READ_L_HIP']
+        elif read_state == serial_read_states['READ_L_HIP']:
+          data_packet['l_hip'] += decoded_data
+          calculated_checksum += decoded_data
+          data_byte_counter -= 1
+          if data_byte_counter == 0:
+            data_byte_counter = self.data_byte_length
+            read_state = serial_read_states['READ_L_KNEE']
+        elif read_state == serial_read_states['READ_L_KNEE']:
+          data_packet['l_knee'] += decoded_data
+          calculated_checksum += decoded_data
+          data_byte_counter -= 1
+          if data_byte_counter == 0:
+            data_byte_counter = self.data_byte_length
+            read_state = serial_read_states['READ_R_HIP']
+        elif read_state == serial_read_states['READ_R_HIP']:
+          data_packet['r_hip'] += decoded_data
+          calculated_checksum += decoded_data
+          data_byte_counter -= 1
+          if data_byte_counter == 0:
+            data_byte_counter = self.data_byte_length
+            read_state = serial_read_states['READ_R_KNEE']
+        elif read_state == serial_read_states['READ_R_KNEE']:
+          data_packet['r_knee'] += decoded_data
+          calculated_checksum += decoded_data
+          data_byte_counter -= 1
+          if data_byte_counter == 0:
+            data_byte_counter = self.data_byte_length
+            read_state = serial_read_states['READ_L_SHOULDER']
+        elif read_state == serial_read_states['READ_L_SHOULDER']:
+          data_packet['l_shoulder'] += decoded_data
+          calculated_checksum += decoded_data
+          data_byte_counter -= 1
+          if data_byte_counter == 0:
+            data_byte_counter = self.data_byte_length
+            read_state = serial_read_states['READ_L_ELBOW']
+        elif read_state == serial_read_states['READ_L_ELBOW']:
+          data_packet['l_elbow'] += decoded_data
+          calculated_checksum += decoded_data
+          data_byte_counter -= 1
+          if data_byte_counter == 0:
+            data_byte_counter = self.data_byte_length
+            read_state = serial_read_states['READ_R_SHOULDER']
+        elif read_state == serial_read_states['READ_R_SHOULDER']:
+          data_packet['r_shoulder'] += decoded_data
+          calculated_checksum += decoded_data
+          data_byte_counter -= 1
+          if data_byte_counter == 0:
+            data_byte_counter = self.data_byte_length
+            read_state = serial_read_states['READ_R_ELBOW']
+        elif read_state == serial_read_states['READ_R_ELBOW']:
+          data_packet['r_elbow'] += decoded_data
+          calculated_checksum += decoded_data
+          data_byte_counter -= 1
+          if data_byte_counter == 0:
+            data_byte_counter = self.data_byte_length
+            read_state = serial_read_states['READ_AT_GOAL']
```

Comment:
Could be replaced with something like this?
```python
for cmd, state in serial_read_states.items():
  if read_state == state:
    data_packet[cmd[5:].lower()] += decoded_data
    calculated_checksum += decoded_data
    data_byte_counter -= 1

    if data_byte_counter == 0:
      data_byte_counter = self.data_byte_length
      read_state = serial_read_states[serial_read_states.index(cmd) + 1]

    break
```

idk about the preamble and the checksum, but I believe that should encompass all of the logic for the sensor readings.

Reply:
I'm a fan of this idea. Will look into implementing it.

Reply:
I was looking into this for ros2sim. When I implement that, we can just use it here. So don't worry about this

Reply:
Sounds good.